### PR TITLE
src/pages: add stepper component for registration process

### PR DESCRIPTION
**Pull request by maceksimon**
* Add `RegisterChallengePage` page
* Use `q-stepper` component to scaffold first 4 steps
* Add custom styles and icons
* Add tests for page

### CODE REVIEW

File: test/cypress/e2e/register_challenge.spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+          cy.dataCy('contact-form-subject')
+            .find('.q-field__control')
+            .should('not.have.class', 'text-negative');
+          cy.dataCy('dialog-content').scrollTo('bottom');
+          cy.dataCy('contact-form-submit').should('be.visible').click();
+          cy.dataCy('dialog-content').scrollTo('top');
+          cy.dataCy('contact-form-message')
+            .find('.q-field__messages')
+            .should('be.visible')
+            .should('contain', i18n.global.t('index.contact.messageRequired'));
```
Comment (tmszi) [inline test/cypress/e2e/register_challenge.spec.cy.js:85]: `.and()`

File: test/cypress/e2e/register_challenge.spec.cy.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            .should('have.class', 'text-negative');
+          cy.dataCy('contact-form-message-input')
+            .should('be.visible')
+            .type('what is the minimum distance to ride to work?');
+          cy.dataCy('dialog-content').scrollTo('bottom');
+          cy.dataCy('contact-form-submit').should('be.visible').click();
+          cy.dataCy('contact-form-email')
+            .find('.q-field__messages')
+            .should('be.visible')
+            .should('contain', i18n.global.t('index.contact.emailRequired'));
```
Comment (tmszi) [inline test/cypress/e2e/register_challenge.spec.cy.js:97]: `and()`

File: test/cypress/e2e/register_challenge.spec.cy.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+          cy.dataCy('switcher-' + initialActiveLocale)
+            .find('.q-btn')
+            .should('have.class', 'bg-secondary');
+
+          locales.forEach((locale) => {
+            if (locale !== initialActiveLocale) {
+              // changing the language
+              cy.dataCy('switcher-' + locale)
+                .should('exist')
+                .should('be.visible')
```
Comment (tmszi) [inline test/cypress/e2e/register_challenge.spec.cy.js:125]: `.and()`

File: test/cypress/e2e/register_challenge.spec.cy.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+              .should('be.equal', 38);
+            // test icon src
+            cy.dataCy('step-1')
+              .find('img')
+              .should('have.attr', 'src', 'src/assets/svg/numeric-1-fill.svg');
+            // display title
+            cy.dataCy('step-1')
+              .find('.q-stepper__title')
+              .should('be.visible')
+              .should(
```
Comment (tmszi) [inline test/cypress/e2e/register_challenge.spec.cy.js:212]: `.and()`

File: test/cypress/e2e/register_challenge.spec.cy.js [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        });
+      cy.dataCy('list-faq-list')
+        .find('.q-expansion-item')
+        .each(($element) => {
+          cy.wrap($element).should('be.visible');
+          cy.wrap($element).click();
+          cy.wrap($element)
+            .find('.q-card__section')
+            .should('be.visible')
+            .should('not.be.empty');
```
Comment (tmszi) [inline test/cypress/e2e/register_challenge.spec.cy.js:24]: `.and()`

File: test/cypress/e2e/register_challenge.spec.cy.js (hunk 1 above)
Comment (maceksimon) [reply test/cypress/e2e/register_challenge.spec.cy.js:85]: Fixed.

File: test/cypress/e2e/register_challenge.spec.cy.js (hunk 2 above)
Comment (maceksimon) [reply test/cypress/e2e/register_challenge.spec.cy.js]: Fixed.

File: test/cypress/e2e/register_challenge.spec.cy.js (hunk 3 above)
Comment (maceksimon) [reply test/cypress/e2e/register_challenge.spec.cy.js:125]: Fixed.

File: test/cypress/e2e/register_challenge.spec.cy.js (hunk 4 above)
Comment (maceksimon) [reply test/cypress/e2e/register_challenge.spec.cy.js]: Fixed.

File: test/cypress/e2e/register_challenge.spec.cy.js (hunk 5 above)
Comment (maceksimon) [reply test/cypress/e2e/register_challenge.spec.cy.js]: Fixed.

File: test/cypress/e2e/register_challenge.spec.cy.js [hunk 6]
```diff
@@ -0,0 +1,298 @@
+describe('Register Challenge page', () => {
+  context('desktop', () => {
+    beforeEach(() => {
+      cy.visit('/#/register-challenge');
+      cy.viewport('macbook-16');
+    });
+
+    it('allows user to display help dialog and read all FAQ items', () => {
```
Comment (tmszi) [inline test/cypress/e2e/register_challenge.spec.cy.js:8]: Here line 8-102 are the duplicate code tests. Will it be possible to write one basic test (help dialog) and import it here?

File: test/cypress/e2e/register_challenge.spec.cy.js [hunk 7]
```diff
@@ -0,0 +1,298 @@
+describe('Register Challenge page', () => {
+  context('desktop', () => {
+    beforeEach(() => {
+      cy.visit('/#/register-challenge');
```
Comment (tmszi) [inline test/cypress/e2e/register_challenge.spec.cy.js:4]: Use named route here, not hardcoded URL string path, please.

Comment (tmszi) [reply test/cypress/e2e/register_challenge.spec.cy.js:4]: Solution:

1. Create ./src/router/routes_conf.ts file

This is example config for register-challenge router path, you must extend `routesConf` object with another router paths e.g. home and etc.

```ts
const routesConf: object = {
    'register-challenge': {
       path: '/register-challenge',
       children: {
         name: 'register-challenge',
       },
    },
};

export { routesConf };
```

2. Adjust ./src/router/routes.ts file

```ts
import { routesConf } from './routes_conf';

...

  {
    path: routesConf['register-challenge']['path'],
    component: () => import('layouts/LoginRegisterLayout.vue'),
    children: [
      {
        path: '',
        component: () => import('pages/RegisterChallengePage.vue'),
        name: routesConf['register-challenge']['children']['name'],
      },
    ],
  },
```

3. Adjust e2e test file ./test/cypress/e2e/register_challenge.spec.cy.js

```js
import { routesConf } from '../../../src/router/routes_conf';

describe('Register Challenge page', () => {
  context('desktop', () => {
    beforeEach(() => {
      cy.visit(routesConf['register-challenge']['path']);
      cy.viewport('macbook-16');
    });
```

File: test/cypress/e2e/register_challenge.spec.cy.js (hunk 6 above)
Comment (maceksimon) [reply test/cypress/e2e/register_challenge.spec.cy.js]: Based on a previous discussion, I have moved these tests into the component.
Tests which will be share but cannot be moved into component (switching language) are extracted into `commonTests.js`.

File: test/cypress/e2e/register_challenge.spec.cy.js (hunk 7 above)
Comment (maceksimon) [reply test/cypress/e2e/register_challenge.spec.cy.js]: Updated.
We still need to prepend `'#'` to navigate to the correct route, but the route is dynamic.
I will update this for other E2E test in a separate PR.

File: test/cypress/support/commonTests.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        cy.dataCy('switcher-' + initialActiveLocale)
+          .find('.q-btn')
+          .should('have.class', 'bg-secondary');
+
+        locales.forEach((locale) => {
+          if (locale !== initialActiveLocale) {
+            // changing the language
+            cy.dataCy('switcher-' + locale)
+              .should('exist')
+              .should('be.visible')
```
[the remark this answers is not in the document]
Comment (maceksimon) [reply test/cypress/support/commonTests.js]: Fixed.

**Pull request merged**